### PR TITLE
docs: updates docstrings for the CLI commands

### PR DESCRIPTION
**Pull request by pallabpain**
### Description

Adds extensive documentation for the CLI commands.

### CODE REVIEW

**Review by pallabpain — COMMENTED** [review]
TOD List
1. Update README and copy the same context in sphinx index.
2. Use the new rapyuta.io logo.

File: riocli/deployment/util.py [hunk 1]
```diff
@@ -77,10 +75,11 @@ def get_deployment_guid(client: Client, name: str) -> str:
 def get_deployment_name(client: Client, guid: str) -> str:
```
Comment (ankitrgadiya) [inline riocli/deployment/util.py:75]: question: Is this function still used?

File: riocli/deployment/wait.py [hunk 2]
```diff
@@ -32,8 +32,10 @@ def wait_for_deployment(
         deployment_name: str,
         spinner=None,
 ) -> None:
-    """
-    Wait until the deployment succeeds/fails
+    """Wait until the deployment succeeds/fails
+
+    This command is useful in scripts or automation when you
+    explicitly want to wait for the deployment to complete.
```
Comment (ankitrgadiya) [inline riocli/deployment/wait.py:38]:
```suggestion
    explicitly want to wait for the deployment to succeed.
```

File: riocli/device/delete.py [hunk 3]
```diff
@@ -50,8 +50,34 @@ def delete_device(
...
+
+      $ rio device delete DEVICE_NAME --force
+
+      Delete all device in the project
+
+      $ rio device delete --all
+
+      Delete devices using regex pattern
+
+      $ rio device delete "DISK.*"
```
Comment (ankitrgadiya) [inline riocli/device/delete.py:80]:
```suggestion
      $ rio device delete "DEVICE.*"
```

File: riocli/device/metric.py [hunk 4]
```diff
@@ -32,8 +32,10 @@
     help_options_color='green',
 )
 def device_metrics():
-    """
-    Device Metrics
+    """Interact with device metrics.
```
Comment (ankitrgadiya) [inline riocli/device/metric.py:35]: todo: These APIs are unsupported now. We can remove these commands.

File: riocli/device/onboard.py [hunk 5]
```diff
@@ -21,12 +21,11 @@
...
 @name_to_guid
 def device_onboard(device_name: str, device_guid: str) -> None:
-    """
-    Generates the on-boarding script for the Device
-
-    To onboard the current device, run the following
+    """Generates the on-boarding script for a device.
 
-        exec $(rio device onboard {device-name})
+    You are supposed to run the script on the device to onboard it.
```
Comment (ankitrgadiya) [inline riocli/device/onboard.py:26]:
```suggestion
    You need to run the script on the device to onboard it.
```

File: riocli/device/tools/rapyuta_logs.py [hunk 6]
```diff
@@ -25,6 +25,11 @@
 @click.argument('device-name', type=str)
 @name_to_guid
 def rapyuta_agent_logs(device_name: str, device_guid: str) -> None:
+    """Stream rapyuta agent logs from the device.
```
Comment (ankitrgadiya) [inline riocli/device/tools/rapyuta_logs.py:28]: thought: Maybe we can consider updating this command to stream deviceedge's logs now.

**Review by ankitrgadiya — CHANGES_REQUESTED** [review]

File: riocli/deployment/util.py (hunk 1 above)
Comment (pallabpain) [reply riocli/deployment/util.py:75]: Yes, in the name to guid decorator. It uses the v2 client.

File: riocli/device/metric.py (hunk 4 above)
Comment (pallabpain) [reply riocli/device/metric.py]: Done.

File: riocli/device/tools/rapyuta_logs.py (hunk 6 above)
Comment (pallabpain) [reply riocli/device/tools/rapyuta_logs.py:28]: Yes. Let's create a task for it and implement it in another PR.

**Review by pallabpain — COMMENTED** [review]
@ankitrgadiya I have addressed the comments. Please take a look.

**Review by ankitrgadiya — APPROVED** [review]

**Pull request merged**